### PR TITLE
go-1.18:bump go to 1.18 and controller-gen v0.10.0

### PR DESCRIPTION
**Pull request by ajayk**
- Update to go 1.18
- Upgrade controller-gen to v0.10.0 , cleans up the crd status field deletion hacks

### CODE REVIEW

**Comment by pintohutch** [comment]
Thanks for the PR @ajayk. I'll take on reviewing this!

File: hack/presubmit.sh [hunk 1]
```diff
@@ -91,12 +91,9 @@ update_crdgen() {
...
     sed -i '0,/---/{/---/d}' $i
-    # Currently controller-gen regenerates the status section of the CRD, which is
-    # not ideal.
-    # There is an open issue: https://github.com/kubernetes-sigs/controller-tools/issues/456.
-    # Until then, we manually delete the status field in the generated CRD yamls here.
-    sed -i '/^status:*/,$d' $i
-    echo "$(cat ${SCRIPT_ROOT}/hack/boilerplate.txt)$(cat $i)" > $i
+    # removed the crd status hack , see https://github.com/kubernetes-sigs/controller-tools/pull/630
+    echo "$(cat $i)" > $i
+    echo -e "$(cat ${SCRIPT_ROOT}/hack/boilerplate.txt)\n$(cat $i)" > $i
```
Comment (TheSpiritXIII) [inline hack/presubmit.sh:96]: Why have this code at all anymore?

Comment (pintohutch) [reply hack/presubmit.sh:96]: Yea - what happens if you remove this for loop altogether?

File: go.mod [hunk 2]
```diff
@@ -37,6 +37,88 @@ require (
 	sigs.k8s.io/yaml v1.3.0
 )
 
+require (
```
Comment (pintohutch) [inline go.mod:40]: It's interesting this was added in the go 1.18 upgrade. AFAICT, this separate block for indirect dependencies was [introduced](https://go.dev/doc/go1.17#go-command) in go 1.17...

**Review by pintohutch — COMMENTED** [review]
Some comments

File: hack/presubmit.sh (hunk 1 above)
Comment (ajayk) [reply hack/presubmit.sh:96]: if er remove the for loop we loose the LICENSE header on the yamls 
https://gist.github.com/ajayk/ffea89d2d25f61d8668984e597c3223d

@TheSpiritXIII , @pintohutch

Comment (pintohutch) [reply hack/presubmit.sh:96]: Ah yea that's right. We want the license headers on the individual files as well as the final combined result. Thanks for confirming!

**Review by pintohutch — APPROVED** [review]
LGTM!

File: go.mod (hunk 2 above)
Comment (ajayk) [reply go.mod:40]: Yep , on the go.mod we jumped from 1.16 to 1.18  :-)

Comment (pintohutch) [reply go.mod:40]: 🤦

Comment (ajayk) [reply go.mod:40]: the Other docker images were on 1.17  !! , now everything is consistent , 
is it the best practice to be on the latest go compiler for builds and the go version lags one version on the module?

Comment (pintohutch) [reply go.mod:40]: no - this was an oversight. thanks for correcting!

**Comment by pintohutch** [comment]
@ajayk can you rebase and commit to resolve merge conflicts?

**Comment by ajayk** [comment]
> @ajayk can you rebase and commit to resolve merge conflicts?

Done ... Do you need a single commit ?

**Comment by pintohutch** [comment]
Nope - just squashed on merge. Thanks for the contribution!

**Pull request merged**